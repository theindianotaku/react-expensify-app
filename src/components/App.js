import React, { Component } from 'react';


class App extends Component {
  render() {
    return (
      <div>
        <p>Hello there! Expensify here.</p>
      </div>
    );
  }
}

export default App;
