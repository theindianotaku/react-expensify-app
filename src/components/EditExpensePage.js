import React from 'react';

const EditExpensePage = (props) => (
  <div>
    <p>Hello! Edit page of id {props.match.params.id} here here.</p>
  </div>
);

export default EditExpensePage;
