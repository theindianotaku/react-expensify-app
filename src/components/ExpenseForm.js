import React, { Component } from 'react';
import moment from 'moment';
import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';

class ExpenseForm extends Component {
  constructor (props) {
    super(props);
    this.state = {
      amount: props.expense ? (props.expense.amount/100) : '',
      createdAt: props.expense? moment(props.expense.createdAt) : moment(),
      description: props.expense ? props.expense.description : '',
      note: props.expense ? props.expense.note : '',
      focused: false,
      error: ''
    };
  }

  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({
      description
    }));
  }

  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({note}));
  }

  onAmountChange = (e) => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  }

  onDateChange = (createdAt) => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  }

  onFocusChange = ({ focused }) => {
    this.setState(() => ({ focused }));
  }

  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.description || !this.state.amount) {
      this.setState(() => ({
        error: 'Please provide description and amount.'
      }));
    } else {
      this.setState(() => ({error: ''}));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };

  render() {
    return (
      <form
        action=""
        className="form"
        onSubmit={this.onSubmit}
      >
        {this.state.error && <p className="form__error">{this.state.error}</p>}

        <input
          type="text"
          placeholder="Description"
          className="text-input"
          autoFocus
          value={this.state.description}
          onChange={this.onDescriptionChange}
        />
        <input
          type="text"
          placeholder="Amount"
          className="text-input"
          value={this.state.amount}
          onChange={this.onAmountChange}
        />

        <SingleDatePicker
          date={this.state.createdAt}
          onDateChange={this.onDateChange}
          focused={this.state.focused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false }
        />

        <textarea
          className="textarea"
          placeholder="Add a note for your expense (optional)"
          value={this.state.note}
          onChange={this.onNoteChange}
        >
        </textarea>
        <button className="button dark" type="Submit">Save Expense</button>
      </form>
    );
  }
}

export default ExpenseForm;
