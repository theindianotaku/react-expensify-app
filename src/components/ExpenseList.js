import React from 'react';

const ExpenseList = () => (
  <div>
    Expense List
  </div>
);

export default ExpenseList;
