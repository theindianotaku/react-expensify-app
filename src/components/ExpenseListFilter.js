import React, { Component } from 'react';
import { connect } from 'react-redux';
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilter extends Component {
  state = {
    calendarFocussed: null
  };

  onDatesChange = ({startDate, endDate}) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChange = (calendarFocussed) => {
    this.setState(() => ({ calendarFocussed }));
  };

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  };

  render () {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              placeholder="Search expenses"
              className="text-input"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
          <div className="input-group__item">
            <select
              className="select"
              onChange={this.onSortChange}
            >
              <option value="date">Date</option>
              <option value="amount">Amount</option>
            </select>
          </div>
          <div className="input-group__item">
            <DateRangePicker
              startDate = {this.props.filters.startDate}
              endDate = {this.props.filters.endDate}
              onDatesChange = {this.onDatesChange}
              focusedInput = {this.state.calendarFocussed}
              onFocusChange = {this.onFocusChange}
              showClearDates = {true}
              numberOfMonths = {1}
              isOutsideRange={() => false}
            />
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilter);
