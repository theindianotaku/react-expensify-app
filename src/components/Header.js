import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';

import { startLogout } from '../thunks/auth';

export const Header = ({ startLogout }) => (
  <header>
    <span>Expensify</span> |
    <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink> |
    <NavLink to="/create" activeClassName="is-active">Create</NavLink> |
    <NavLink to="/help" activeClassName="is-active">Help</NavLink> | 
    <button onClick={startLogout} >Logout</button>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
