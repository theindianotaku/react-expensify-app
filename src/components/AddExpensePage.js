import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { startAddExpense } from '../thunks/expenses';
import ExpenseForm from './ExpenseForm';

export class AddExpensePage extends Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <p className="page-header__title">Add Expense</p>
            <div className="page-header__actions">
              <Link className="button simple valign-wrapper" to="/dashboard">Go back</Link>
            </div>
          </div>
        </div>
        <div className="content-container m-b-l">
          <ExpenseForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
