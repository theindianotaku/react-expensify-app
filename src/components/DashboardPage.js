import React, { Component } from 'react';

import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilter';
import ExpenseSummary from './ExpenseSummary';

class DashboardPage extends Component {
  render() {
    return (
      <div>
        <ExpenseSummary />
        <ExpenseListFilter />
        <ExpenseList />
      </div>
    );
  }
}

export default DashboardPage;
