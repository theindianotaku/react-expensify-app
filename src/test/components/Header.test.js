import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

let startLogout, wrapper;

beforeEach(() => {
  startLogout = jest.fn();
  wrapper = shallow(<Header startLogout={startLogout} />);
});

test('should render Header component', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
  wrapper.find('button').simulate('click');
  expect(startLogout).toHaveBeenCalled();
});
