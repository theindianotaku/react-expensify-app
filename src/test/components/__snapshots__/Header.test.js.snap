// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component 1`] = `
<header>
  <span>
    Expensify
  </span>
   |
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
   |
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create
  </NavLink>
   |
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    Help
  </NavLink>
   |
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
